### PR TITLE
fix(curriculum): moved `z-index` to new step

### LINKED ISSUE #48874: z-index mentioned in passing in balance sheet before being introduced in a later project (picasso painting)
### Describe the Issue

In the balance sheet project step 42:
https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-more-about-css-pseudo-selectors-by-building-a-balance-sheet/step-42
The following instruction is given:
Ensure your years do not get hidden by setting a z-index of 999.

No explanation is given as to what z-index is used for other than this one line.

Not until we get to step 10 (https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-intermediate-css-by-building-a-picasso-painting/step-10) of the Picasso painting project do we see a full explanation for it.
Perhaps the explanation should be moved from the Picasso project to the Balance Sheet project?

### Affected Page

see writeup above

### Your code

```

```

### Expected behavior

Since we are asking people to set the z-index in step 42 of the balance sheet project, and since an explanation already exists for what z-index is used for but in a -later- project, maybe we should move that explanation to step 42 of the balance sheet project or at least give a longer explanation of what z-index is supposed to be doing.

### Screenshots

_No response_

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leslieyip02**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #48874 

I removed `z-index` from `#years` from step 42 onwards.

### CODE REVIEW

**Review by Sboonny — COMMENTED** [review]
I didn't mention removing it from the solution too, sorry about that 

Here is the file https://github.com/leslieyip02/freeCodeCamp/blob/67436e6c9b78b77cabf5b534214f2e09fd84110c/curriculum/challenges/english/14-responsive-web-design-22/learn-more-about-css-pseudo-selectors-by-building-a-balance-sheet/6201a59be346d399c21d10b1.md#L493

----

It shouldn't affect anything keeping it, but the files will cleaner without it.

**Review by Sboonny — COMMENTED** [review]
#### :sparkles: :tada: **AWESOME!** :tada: :sparkles:

 Hi @leslieyip02, thanks for this pull request and for contributing to the code-base for the first time. We are looking forward to more contributions from you in the future.

 Cheers & happy contributing!

**Review by Sboonny — APPROVED** [review]

**Comment by bbsmooth** [comment]
Are we sure we don't need this? The years heading is using fixed positioning. When I remove `z-index` then the table content overlaps the heading.

**Comment by leslieyip02** [comment]
I've modified the CSS for the solution to prevent the table from overlapping with the headers. Could someone check if these changes are ok before I modify the rest of the files?

**Comment by Sboonny** [comment]
Thank you for the hard work @leslieyip02, I have pushed a commit of the requested change because you shouldn't do more for mistake on our part, we should have done our due diligence.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-more-about-css-pseudo-selectors-by-building-a-balance-sheet/64076bbeba941114b0027b6a.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+---
+id: 64076bbeba941114b0027b6a
+title: Step 43
+challengeType: 0
+dashedName: step-43
+---
+
+# --description--
+
+Adding position `sticky` moved the element into its own stack, to ensure your `#years` don't get hidden by different stacks add `z-index` property set to `999` in `#years` CSS rule.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-more-about-css-pseudo-selectors-by-building-a-balance-sheet/64076bbeba941114b0027b6a.md:10]:
```suggestion
Adding position `sticky` moved the element into its own stack. To ensure your `#years` element does not get hidden by different stacks, add a `z-index` property set to `999` in the `#years` rule.
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-more-about-css-pseudo-selectors-by-building-a-balance-sheet/64076bbeba941114b0027b6a.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+# --hints--
+
+Your `#years` selector should have a `z-index` property set to `999`.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('#years')?.getPropertyValue('z-index') === '999');
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-more-about-css-pseudo-selectors-by-building-a-balance-sheet/64076bbeba941114b0027b6a.md:17]:
```suggestion
assert.equal(new __helpers.CSSHelp(document).getStyle('#years')?.getPropertyValue('z-index'), '999');
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
This looks good to me 👍

**Pull request merged**